File: ios/Classes/Features/CrashReportCallback.swift
import ADEUMInstrumentation
import Flutter

class CrashCallbackObject: NSObject, ADEumCrashReportCallback {
  func onCrashesReported(_ summaries: [ADEumCrashReportSummary]) {
    guard let channel = SwiftAppDynamicsAgentPlugin.channel else {
      print("SwiftAppDynamicsAgentPlugin channel is nil.")
      return
    }
    
    let serializableSummaries = summaries.map { summary -> [String: String?] in
      return [
        "crashId": summary.crashId,
        "exceptionName": summary.exceptionName,
        "exceptionReason": summary.exceptionReason,
        "signalName": summary.signalName,
        "signalCode": summary.signalCode,
      ]
    }
    
    channel.invokeMethod("onCrashReported", arguments: serializableSummaries)
  }
}

File: android/src/main/kotlin/com/appdynamics/appdynamics_agent/features/CrashReportCallback.kt
/*
 * Copyright (c) 2022. AppDynamics LLC and its affiliates.
 * All rights reserved.
 *
 */

package com.appdynamics.appdynamics_agent.features

import com.appdynamics.eumagent.runtime.CrashReportCallback
import com.appdynamics.eumagent.runtime.CrashReportSummary
import io.flutter.plugin.common.MethodChannel

class CrashCallbackObject(private val channel: MethodChannel) : CrashReportCallback {
    override fun onCrashesReported(summaries: Collection<CrashReportSummary>) {
        val serializableSummaries = summaries.map { summary ->
            mapOf(
                "crashId" to summary.crashId,
                "exceptionName" to summary.exceptionClass,
                "exceptionReason" to summary.exceptionMessage
            )
        }
        channel.invokeMethod("onCrashReported", serializableSummaries)
    }
}